pkg/ctl/delete/cluster.go

```diff
@@ -103,129 +68,10 @@ func doDeleteCluster(cmd *cmdutils.Cmd) error {
-			ctx, cleanup := context.WithTimeout(context.Background(), 10*time.Minute)
-			defer cleanup()
-
-			logger.Info("cleaning up AWS load balancers created by Kubernetes objects of Kind Service or Ingress")
-			if err := elb.Cleanup(ctx, ctl.Provider.EC2(), ctl.Provider.ELB(), ctl.Provider.ELBV2(), clientSet, cfg); err != nil {
-				return err
-			}
-		}
-
-		deleteOIDCProvider := clusterOperable && oidcSupported
-		tasks, err := stackManager.NewTasksToDeleteClusterWithNodeGroups(deleteOIDCProvider, oidc, kubernetes.NewCachedClientSet(clientSet), cmd.Wait, func(errs chan error, _ string) error {
-			logger.Info("trying to cleanup dangling network interfaces")
-			if err := ctl.LoadClusterVPC(cfg); err != nil {
-				return errors.Wrapf(err, "getting VPC configuration for cluster %q", cfg.Metadata.Name)
-			}
-
-			go func() {
-				errs <- vpc.CleanupNetworkInterfaces(ctl.Provider.EC2(), cfg)
-				close(errs)
-			}()
-			return nil
-		})
-
-		if err != nil {
-			return err
-		}
-
-		if tasks.Len() == 0 {
-			logger.Warning("no cluster resources were found for %q", meta.Name)
-			return nil
-		}
-
-		logger.Info(tasks.Describe())
-		if errs := tasks.DoAllSync(); len(errs) > 0 {
-			return handleErrors(errs, "cluster with nodegroup(s)")
-		}
-
-		logger.Success("all cluster resources were deleted")
-	}
-
-	{
-		if err := gitops.DeleteKey(cfg); err != nil {
-			return err
-		}
-	}
-
-	return nil
-}
-
-func deleteFargateProfiles(cmd *cmdutils.Cmd, ctl *eks.ClusterProvider) error {
-	awsClient := fargate.NewClientWithWaitTimeout(
-		cmd.ClusterConfig.Metadata.Name,
-		ctl.Provider.EKS(),
-		cmd.ProviderConfig.WaitTimeout,
-	)
-	profileNames, err := awsClient.ListProfiles()
+	cluster, err := cluster.New(cfg, ctl)
 	if err != nil {
-		if fargate.IsUnauthorizedError(err) {
-			logger.Debug("Fargate: unauthorized error: %v", err)
-			logger.Info("either account is not authorized to use Fargate or region %s is not supported. Ignoring error",
-				cmd.ClusterConfig.Metadata.Region)
-			return nil
-		}
 		return err
 	}
 
-	// Linearise the deleting of Fargate profiles by passing as the API
-	// otherwise errors out with:
-	//   ResourceInUseException: Cannot delete Fargate Profile ${name2} because
-	//   cluster ${clusterName} currently has Fargate profile ${name1} in
-	//   status DELETING
-
-	for _, profileName := range profileNames {
-		logger.Info("deleting Fargate profile %q", *profileName)
-		// All Fargate profiles must be completely deleted by waiting for the deletion to complete, before deleting
-		// the cluster itself, otherwise it can result in this error:
-		//   Cannot delete because cluster <cluster> currently has Fargate profile <profile> in status DELETING
-		if err := awsClient.DeleteProfile(*profileName, true); err != nil {
-			return err
-		}
-		logger.Info("deleted Fargate profile %q", *profileName)
-	}
-	logger.Info("deleted %v Fargate profile(s)", len(profileNames))
-	return nil
+	return cluster.Delete(time.Second*20, cmd.Wait)
```

Comment:
Possible to have this timeout (which i think is also used somewhere else) saved and named somewhere?

Reply:
I did try to find somewhere appropriate for this, but couldn't find one. The regular `WaitTimeout` is the max duration the command should run for and is set on the cli by `--timeout 25m0s`, because its configurable it lives on the `ctl.Provider`. This value here is simply the interval between polling for, made configurable for unit testing (it was this or injecting `time` 😬 ). I don't believe this belongs on the `ctl.Provider`, so I'm not sure where to put it 🤷

Reply:
Yeh idk either 🤷‍♀️ . That I said "whatever it is" above, but maybe we can live without it named. Doesn't feel great tho 😬